### PR TITLE
feat: Add `KeyPairResourcePolicy.max_session_lifetime`

### PR DESCRIPTION
**Pull request by achimnol**
resolves lablup/backend.ai-internal#46

### CODE REVIEW

**Comment by achimnol** [comment]
TODO: add a timer to periodically check expired sessions

File: src/ai/backend/manager/idle.py [hunk 1]
```diff
@@ -343,13 +370,29 @@ async def check_session(self, session: Row, dbconn: SAConnection) -> bool:
...
+
+    async def populate_config(self, config: Mapping[str, Any]) -> None:
+        pass
+
+    async def check_session(self, session: Row, dbconn: SAConnection, policy: Row) -> bool:
+        now = await dbconn.execute(sa.select(sa.func.now()))
+        if policy["max_session_lifetime"] >= 0:
+            # TODO: once per-status time tracking is implemented, let's change created_at
+            #       to the timestamp when the session entered PREPARING status.
+            if now - session["created_at"] >= policy["max_session_lifetime"]:
```
Comment (adrysn) [inline src/ai/backend/manager/idle.py:385]: If `max_session_lifetime` is zero, I suggest interpreting it as infinity. Also, these lines raise exceptions due to calculation type-mismatch issues. These might be updated to something like below:

```python
now = await dbconn.scalar(sa.select(sa.func.now()))
if policy["max_session_lifetime"] > 0:
    # TODO: once per-status time tracking is implemented, let's change created_at
    #       to the timestamp when the session entered PREPARING status.
    if now - session["created_at"] >= timedelta(seconds=policy["max_session_lifetime"]):
```

**Review by adrysn — CHANGES_REQUESTED** [review]
I found an error in `SessionlifetimeChecker. Other parts seem to be working well in my test environment.

**Review by adrysn — APPROVED** [review]
Tested and LGTM.

**Pull request merged**